documentation/en/block-validation.md

```diff
@@ -0,0 +1,135 @@
+# Incoming block validations
+
+This document reviews the code flow that takes place inside the full node after receiving a new block from the GossipSub `/fil/blocks` topic and traces all of its protocol-related validation logic. We do not include validation logic *inside* the VM, the analysis stops at `(*VM).Invoke()`. The `V:` tag explicitly signals validations throughout the text.
+
+## `modules.HandleIncomingBlocks()`
+
+We subscribe to the `/fil/blocks` PubSub topic to receive external blocks from peers in the network and register a block validator that operates at the PubSub (`libp2p` stack) level, validating each PubSub message containing a Filecoin block header.
+
+`V:` PubSub message is a valid CBOR `BlockMsg`.
+
+`V:` Total messages in block are under `BlockMessageLimit`.
+
+`V:` Aggregate message CIDs, encapsulated in the `MsgMeta` structure, serialize to the `Messages` CID in the block header (`ValidateMsgMeta()`).
+
+`V:` Miner `Address` in block header is present and corresponds to a public-key address in the current chain state.
+
+`V:` Block signature (`BlockSig`) is present and belongs to the public-key address retrieved for the miner (`CheckBlockSignature()`).
```

Comment:
we might also want to mention the logic about chain syncing here: if the chain is not synced (within 6 hours) and the miner is unknown, the block is ignored. If the chain is synced and the miner is unknown the block is rejected.

Reply:
ok, makes sense.